Quantum-Control-Applications-QuAM/Superconducting/configuration/generate_superconducting_quam.py

```diff
@@ -0,0 +1,312 @@
+                intermediate_frequency=50 * u.MHz,
+                depletion_time=1 * u.us,
+            ),
+        )
+        machine.qubits[transmon.name] = transmon
+        machine.active_qubit_names.append(transmon.name)
+
+        add_default_transmon_pulses(transmon)
+
+        RF_output = transmon.xy.frequency_converter_up
+        RF_output.channel = transmon.xy.get_reference()
+        RF_output.output_mode = "always_on"
+        RF_output.LO_frequency = 7 * u.GHz
+        print(f"Please set the LO frequency of {RF_output.get_reference()}")
+
+    # Only set resonator RF outputs once
+    RF_output_resonator = transmon.resonator.frequency_converter_up
+    RF_output_resonator.channel = transmon.resonator.get_reference()
+    RF_output_resonator.output_mode = "always_on"
+    RF_output_resonator.LO_frequency = 4 * u.GHz
+    print(f"Please set the LO frequency of {RF_output_resonator.get_reference()}")
+
+    RF_input_resonator = transmon.resonator.frequency_converter_down
+    RF_input_resonator.channel = transmon.resonator.get_reference()
+    RF_input_resonator.LO_frequency = 4 * u.GHz
+    print(f"Please set the LO frequency of {RF_input_resonator.get_reference()}")
+
+    return machine
+
+
+def _example_wiring(num_qubits: int):
+    """
+    An example wiring is provided in the following physical order on the
+    numbered channels of the OPX+ and Octave:
+
+    1. Assigns wiring for a single feed-line for the resonator.
+    2. Assigns XY wiring (I and Q) consecutively for each qubit.
+    3. Assigns Z wiring consecutively for each qubit.
+
+    Notes:
+    - Requires multiple OPX+ after 2 qubits.
+    - Requires multiple octaves after 4 qubits.
+    """
+    xy_lines, flux_lines, res_lines = [], [], []
+    num_feedlines = 1
+    num_opx_chs = 10
+    for q_idx in range(num_qubits):
+        xy_idx = 2 * num_feedlines + 2 * q_idx
+        z_idx = 2 * num_feedlines + 2 * num_qubits + q_idx
+
+        xy_ch = xy_idx % num_opx_chs + 1
+        z_ch = z_idx % num_opx_chs + 1
+
+        xy_opx = xy_idx // num_opx_chs + 1
+        z_opx = z_idx // num_opx_chs + 1
+
+        xy_octave = xy_opx
+        xy_octave_ch = (xy_ch + 1) // 2
+
+        xy_lines.append((f"con{xy_opx}", xy_ch, f"octave{xy_octave}", xy_octave_ch))
+        flux_lines.append((f"con{z_opx}", z_ch))
+        res_lines.append((f"con1", 1, "octave1", 1))
+
+    return xy_lines, flux_lines, res_lines
+
+
+if __name__ == "__main__":
+    folder = Path(__file__).parent
+    quam_folder = folder / "quam_state"
+
+    machine = create_quam_superconducting(num_qubits=5)
+    machine.save(quam_folder, content_mapping={"wiring.json": {"wiring", "network"}})
+
+    qua_file = folder / "qua_config.json"
+    qua_config = machine.generate_config()
+    json.dump(qua_config, qua_file.open("w"), indent=4)
+
+    quam_loaded = QuAM.load(quam_folder)
+    qua_file_loaded = folder / "qua_config2.json"
```

Comment:
I understand the need to see whether the loaded QuAM can also generate a config, but what about saving to the same file so we don't get people asking why there is also `qua_config2.json`?